File: php/src/Advertisements/User/Application/Command/DisableMember/DisableMemberUseCase.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\User\Application\Command\DisableMember;

use Demo\App\Advertisements\Shared\Domain\ValueObjects\UserId;
use Demo\App\Advertisements\User\Domain\Exceptions\AdminWithIncorrectCivicCenterException;
use Demo\App\Advertisements\User\Domain\Exceptions\MemberDoesNotExistsException;
use Demo\App\Advertisements\User\Domain\Exceptions\UserNotFoundException;
use Demo\App\Advertisements\User\Domain\UserRepository;
use Exception;

final class DisableMemberUseCase
{
    public function __construct(private UserRepository $userRepository)
    {
    }

    /**
     * @throws Exception
     */
    public function execute(DisableMemberCommand $command): void
    {
        $adminUser = $this->userRepository->findAdminById(new UserId($command->securityUserId));
        if (!$adminUser) {
            throw UserNotFoundException::asAdmin();
        }
        $member = $this->userRepository->findMemberById(new UserId($command->memberId));

        if (null === $member) {
            throw MemberDoesNotExistsException::build();
        }

        if (!$adminUser->civicCenterId()->equals($member->civicCenterId())) {
            throw AdminWithIncorrectCivicCenterException::differentCivicCenterFromMember();
        }

        $member->disable();

        $this->userRepository->saveMember($member);
    }
}

File: kotlin/src/main/kotlin/advertisements/user/application/command/disablemember/DisableMemberUseCase.kt
package advertisements.user.application.command.disablemember

import advertisements.shared.value_object.UserId
import advertisements.user.domain.UserRepository
import advertisements.user.domain.exceptions.AdminWithIncorrectCivicCenterException
import advertisements.user.domain.exceptions.MemberDoesNotExistsException
import advertisements.user.domain.exceptions.UserNotFoundException

class DisableMemberUseCase(private val userRepository: UserRepository) {

    fun execute(command: DisableMemberCommand) {
        val adminUser = userRepository.findAdminById(UserId(command.securityUserId))
            ?: throw UserNotFoundException.asAdmin()

        val member = userRepository.findMemberById(UserId(command.memberId))
            ?: throw MemberDoesNotExistsException.build()

        if (!adminUser.civicCenterId().equals(member.civicCenterId())) {
            throw AdminWithIncorrectCivicCenterException.differentCivicCenterFromMember()
        }

        member.disable()

        userRepository.saveMember(member)
    }
}
